rasa/cli/x.py

```diff
@@ -493,28 +493,14 @@ def run_locally(args: argparse.Namespace) -> None:
 
     # noinspection PyBroadException
     try:
-        try:
-            local.main(
-                args,
-                project_path,
-                args.data,
-                token=rasa_x_token,
-                config_path=config_path,
-                domain_path=domain_path,
-            )
-        except TypeError as e:
-            if "domain_path" in str(e):
-                # backwards compatibility with Rasa X versions < 0.35.0
-                # fixes regression https://github.com/RasaHQ/rasa/issues/7592
-                local.main(
-                    args,
-                    project_path,
-                    args.data,
-                    token=rasa_x_token,
-                    config_path=config_path,
-                )
-            else:
-                raise
+        local.main(
+            args,
+            project_path,
+            args.data,
+            token=rasa_x_token,
+            config_path=config_path,
+            domain_path=domain_path,
+        )
```

Comment:
Reason for the change: 3.0 will be supported by 1.x versions of Rasa Enterprise. Probably would have been better to warn the user in advance when introducing the backwards compatibility change. I think it's low risk to remove

Reply:
🤔 looking at the [compatibility matrix](https://rasa.com/docs/rasa-x/changelog/compatibility-matrix) it seems that we can go ahead with the change. 0.35 and below wouldn't work anyway with 3.0. The current discussion is about the latest Rasa X version supporting either 3.0 only or 3.0+2.8 (not 2.2.4).